### PR TITLE
Add policies and logic to handle dnssec for route53 hosted zone

### PR DESCRIPTION
**Pull request by alinabuzachis**
Add policies and logic to handle dnssec for route53 hosted zone

### CODE REVIEW

File: aws/terminator/networking.py [hunk 1]
```diff
@@ -32,6 +44,15 @@ def terminate(self):
                     } for record_set in remove_record_sets]
                 }
             )
+
+        dnssec = self.client.get_dnssec(HostedZoneId=self.id)
+        if dnssec["Status"]["KeySigningKeys"] != []:
```
Comment (gravesm) [inline aws/terminator/networking.py:49]:
```suggestion
        if dnssec["KeySigningKeys"] != []:
```

File: aws/terminator/networking.py [hunk 2]
```diff
@@ -32,6 +44,15 @@ def terminate(self):
...
+            if dnssec["Status"]["ServerSignature"] != "SIGNING":
```
Comment (gravesm) [inline aws/terminator/networking.py:50]:
```suggestion
            if dnssec["Status"]["ServeSignature"] != "SIGNING":
```

File: aws/terminator/networking.py [hunk 3]
```diff
@@ -32,6 +44,15 @@ def terminate(self):
...
+                # Disable DNSSEC for the hosted zone
+                self.client.disable_hosted_zone_dnssec(HostedZoneId=self.id)
+
+            self._handle_ksk(dnssec["Status"]["KeySigningKeys"])
```
Comment (gravesm) [inline aws/terminator/networking.py:54]:
```suggestion
            self._handle_ksk(dnssec["KeySigningKeys"])
```

**Review by gravesm — COMMENTED** [review]
The tests are still failing for me, but it seems like it may be a problem with the key policy. I playing around with it, but couldn't get it to work.

**Comment by alinabuzachis** [comment]
> The tests are still failing for me, but it seems like it may be a problem with the key policy. I playing around with it, but couldn't get it to work.

@gravesm  Do you think "kms:Sign" and "kms:Verify" are also needed?

**Comment by gravesm** [comment]
This is the error I get:

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: botocore.errorfactory.InvalidKMSArn: An error occurred (InvalidKMSArn) when calling the ActivateKeySigningKey operation: The customer managed KMS key with the ARN arn:aws:kms:us-east-1:966509639900:key/b3655f7f-1806-41e3-adcc-f787392c8589 does not grant all the required permissions for DNSSEC usage. Please review the key policy, and verify that you and Route 53 have permissions for the following actions: DescribeKey, GetPublicKey, and Sign. After you make sure that you have the right permissions in place, activate the key-signing key again.
fatal: [testhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):
  File \"<stdin>\", line 121, in <module>
  File \"<stdin>\", line 113, in _ansiballz_main
  File \"<stdin>\", line 61, in invoke_module
  File \"<frozen runpy>\", line 226, in run_module
  File \"<frozen runpy>\", line 98, in _run_module_code
  File \"<frozen runpy>\", line 88, in _run_code
  File \"/tmp/ansible_amazon.aws.route53_ksk_payload_b9imoxgr/ansible_amazon.aws.route53_ksk_payload.zip/ansible_collections/amazon/aws/plugins/modules/route53_ksk.py\", line 360, in <module>
  File \"/tmp/ansible_amazon.aws.route53_ksk_payload_b9imoxgr/ansible_amazon.aws.route53_ksk_payload.zip/ansible_collections/amazon/aws/plugins/modules/route53_ksk.py\", line 344, in main
  File \"/tmp/ansible_amazon.aws.route53_ksk_payload_b9imoxgr/ansible_amazon.aws.route53_ksk_payload.zip/ansible_collections/amazon/aws/plugins/modules/route53_ksk.py\", line 272, in create_or_update
  File \"/tmp/ansible_amazon.aws.route53_ksk_payload_b9imoxgr/ansible_amazon.aws.route53_ksk_payload.zip/ansible_collections/amazon/aws/plugins/modules/route53_ksk.py\", line 216, in activate
  File \"/usr/local/lib/python3.11/dist-packages/botocore/client.py\", line 534, in _api_call
    return self._make_api_call(operation_name, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/usr/local/lib/python3.11/dist-packages/botocore/client.py\", line 976, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.InvalidKMSArn: An error occurred (InvalidKMSArn) when calling the ActivateKeySigningKey operation: The customer managed KMS key with the ARN arn:aws:kms:us-east-1:966509639900:key/b3655f7f-1806-41e3-adcc-f787392c8589 does not grant all the required permissions for DNSSEC usage. Please review the key policy, and verify that you and Route 53 have permissions for the following actions: DescribeKey, GetPublicKey, and Sign. After you make sure that you have the right permissions in place, activate the key-signing key again.
", "module_stdout": "", "msg": "MODULE FAILURE
See stdout/stderr for the exact error", "rc": 1}
```

**Comment by alinabuzachis** [comment]
> botocore.errorfactory.InvalidKMSArn: An error occurred (InvalidKMSArn) when calling the ActivateKeySigningKey operation: The customer managed KMS key with the ARN arn:aws:kms:us-east-1:966509639900:key/b3655f7f-1806-41e3-adcc-f787392c8589 does not grant all the required permissions for DNSSEC usage. Please review the key policy, and verify that you and Route 53 have permissions for the following actions: DescribeKey, GetPublicKey, and Sign. After you make sure that you have the right permissions in place, activate the key-signing key again.

@gravesm Did you added a policy for those permissions to the kms key (something like https://github.com/ansible-collections/amazon.aws/pull/2412/files#diff-c79b524d5d54af3bade314a147303d353c103ff47bd90dd709267e152df756f7R11-R15)?

**Comment by gravesm** [comment]
OK, yeah, looks like we just need to add `kms:Sign` and `kms:Verify`.

**Comment by alinabuzachis** [comment]
> OK, yeah, looks like we just need to add `kms:Sign` and `kms:Verify`.

I added them in aws/policy/security-services.yaml and I had to group some policies because of the file length issue.

**Comment by gravesm** [comment]
Yep, confirm it's working now. Tests still fail but not for permission reasons.

**Review by gravesm — APPROVED** [review]

**Pull request merged**